packages/manager/modules/iam/src/components/actionSelect/actionSelect.controller.js

```diff
@@ -367,19 +367,56 @@ export default class ActionSelectController {
    * @param {Category} category
    * @returns {string}
    */
-  getCategoryLabel({ actions }) {
+  getCategoryLabel({ actions }, searchQuery) {
     const selectionCount = this.constructor.countSelectedActions(actions);
+    const searchCount = this.constructor.countActionsMatchingSearch(
+      actions,
+      searchQuery,
+    );
     const prefix = 'iam_action_select_category_count';
+    if (selectionCount && searchQuery) {
+      if (searchCount <= 1) {
+        return selectionCount === 1
+          ? `${prefix}_selection_one_result_one`
+          : `${prefix}_selection_many_result_one`;
+      }
+      return selectionCount === 1
+        ? `${prefix}_selection_one_result_many`
+        : `${prefix}_selection_many_result_many`;
+    }
     if (selectionCount) {
       return selectionCount === 1
         ? `${prefix}_selection_one`
         : `${prefix}_selection_many`;
     }
+    if (searchQuery) {
+      return searchCount <= 1
+        ? `${prefix}_result_one`
+        : `${prefix}_result_many`;
+    }
     return actions.length === 1
       ? `${prefix}_no_selection_one`
       : `${prefix}_no_selection_many`;
   }
 
+  filterActions(actionTree) {
+    angular.forEach(actionTree.categories, (category) => {
+      // eslint-disable-next-line no-param-reassign
```

Comment:
Use `Array.prototype.map` but dont disable the ESLint rules

Reply:
Hello, Array.map does not solve the issue. 
The problem is that this rule is used to prevent unintentionnaly mutate a parameter. But this is what It is meant for. I want for each category to update the array of objects filtered according to filter, which seems normal to me. As stated here https://stackoverflow.com/questions/35637770/how-to-avoid-no-param-reassign-when-setting-a-property-on-a-dom-object, a solution could be to use an intermediary variable but I is not what I want, since I want to update the original value. The only other viable solution is to disable the rule, which is use pretty often in the codebase. Unless you see another solution which I don't.

Reply:
Please use a temporary variable. Otherwise, you can accept an array as a param and return a new modified array and assign it to whatever variable you want.

Reply:
Done